crates/editor/src/element.rs

```diff
@@ -2287,18 +2289,39 @@ impl EditorElement {
                     }
 
                     if !contains_non_empty_selection {
-                        let origin = point(
-                            layout.hitbox.origin.x,
-                            layout.hitbox.origin.y
-                                + (start_row.as_f32() - scroll_top)
-                                    * layout.position_map.line_height,
-                        );
-                        let size = size(
-                            layout.hitbox.size.width,
-                            layout.position_map.line_height * (end_row - start_row.0 + 1) as f32,
-                        );
-                        let active_line_bg = cx.theme().colors().editor_active_line_background;
-                        cx.paint_quad(fill(Bounds { origin, size }, active_line_bg));
+                        let highlight_h_range =
+                            match EditorSettings::get_global(cx).current_line_highlight {
```

Comment:
Different editors might need different highlight strategies, so I do not think we should hardcode the settings access in the rendering code.

Let's instead have a field in the EditorSnapshot that gets the default from settings, but could be changed programmatically with some `editor.set_current_line_highlight` API.

------------------

Usecase example: deleted diff highlights
![image](https://github.com/zed-industries/zed/assets/2690773/64bc6cf4-982f-4d3c-be15-3a1f3601dec3)

As you can notice, there's currently two line highlights happening, this is due to deleted text being in displayed in another editor.
There's a way to use this editor programmatically:
https://github.com/zed-industries/zed/blob/91b9e4ee9f121d20c2adfbd29fb7a0b5b1b8d614/crates/editor/src/hunk_diff.rs#L583-L588

but no good way to remove the last selection, as may places in the code think that at least one exists, e.g. 

https://github.com/zed-industries/zed/blob/91b9e4ee9f121d20c2adfbd29fb7a0b5b1b8d614/crates/editor/src/selections_collection.rs#L357

https://github.com/zed-industries/zed/blob/91b9e4ee9f121d20c2adfbd29fb7a0b5b1b8d614/crates/editor/src/selections_collection.rs#L248

Thanks to this PR we could use different settings on focused and blurred deleted editors.

Reply:
> Thanks to this PR we could use different settings on focused and blurred deleted editors.

We'll be able to remove the line highligh upon the creation of such a buffer, but it'll come back when the user changes settings. Is there a common way in Zed to deal with such tasks? Some flag that prevents a component from reacting on particular setting change? Some hidden from the users enum value that signals to not change it in `settings_changed()`? Some other solution?

Reply:
Good point, there's no good solution for that yet — editors in editors is a new case (and hopefully, not very widespread and temporary for git).
I can handle it myself somehow (or maybe leave as is for the time being as it's not very frequent to have both deleted hunks toggled and the corresponding settings changed).

Thank you for changing the API though.

Reply:
https://github.com/zed-industries/zed/pull/11772

Here's what I came up with: 
* no `notify` in the editor API is needed, as we might change more things before repainting
* more focus and settings listeners added to toggle the line highlight according to the logic: no highlights in a non-focused deleted diff editor, parent editor's highlights anyway
* one drawback is that the artificial gutter is not highlighted for such editors, but that seems fine for now